[#299] Teams not supported on Hybrid orgs

arlina-espinoza opened the pull request:
Fixes #299 . PR adds an install/runtime requirements check to `apigee_edge_teams` module that verifies if the configured Edge connection is to a Hybrid org, and if so, throws a requirements error. How this works:

- If the Edge connection has already been configured when trying to enable the teams module, and it is connecting to a Hybrid org, it will prevent installing the module and display an error: _Teams are not supported for your Apigee Edge organization._
- If the Edge connection still hasn't been setup when enabling the teams module, installation will proceed. When the connection to a Hybrid org has been configured, an error will be shown on Drupal's status report page: _Teams are not supported for your Apigee Edge organization._

NOTE: Requires https://github.com/apigee/apigee-client-php/pull/93 on the PHP client.

cnovak commented on modules/apigee_edge_teams/apigee_edge_teams.install:
```diff
@@ -0,0 +1,55 @@
+      if ($organization && !OrganizationFeatures::isCompaniesFeatureAvailable($organization)) {
+        $requirements['apigee_edge_teams_not_supported'] = [
+          'title' => t('Apigee Edge Teams'),
+          'description' => t('Teams are not supported for your Apigee Edge organization.'),
```
I know the `isCompaniesFeatureAvailable()` does not make it clear, but we should explain that teams resource does not exist in hybrid so that the user understands why teams is not available, or else you would be left wondering why your org does not support this feature and how to get it enabled.

```suggestion
          'description' => t('Teams are not available for your org because <a href=':url' target='_blank'>Edge company APIs are not supported in Apigee hybrid orgs</a>.', [':url' => 'https://docs.apigee.com/hybrid/compare-hybrid-edge#unsupported-apis'] ),
```

cnovak commented on modules/apigee_edge_teams/apigee_edge_teams.install:
```diff
@@ -0,0 +1,55 @@
+function apigee_edge_teams_requirements($phase) {
+  $requirements = [];
+
+  if ($phase == 'install' || $phase == 'runtime') {
```
If the phase is runtime, here is what I see in status report:

<img width="884" alt="team-error" src="https://user-images.githubusercontent.com/613093/69388660-88dfc380-0c7e-11ea-83b4-19545f61607e.png">

If it is runtime, we should add additional text to tell user what needs to be done:

> The Apigee Edge Teams module functionality will not work when connecting to an Edge hybrid org and should be uninstalled.

arlina-espinoza replied:
I've updated the messages so that on install:
_The Apigee Edge Teams module functionality is not available for your org because [Edge company APIs are not supported in Apigee hybrid orgs](https://docs.apigee.com/hybrid/compare-hybrid-edge#unsupported-apis)._

And on runtime:
_The Apigee Edge Teams module functionality is not available for your org and should be uninstalled, because [Edge company APIs are not supported in Apigee hybrid orgs](https://docs.apigee.com/hybrid/compare-hybrid-edge#unsupported-apis)._

arlina-espinoza commented:
> We need to update the composer.json file to make sure the Apigee client library is updated or the `isCompaniesFeatureAvailable()` function will not exist

I've updated composer.json to require the dev hybrid branch of the Apigee client library, and added an issue to this project to update the composer file once that work gets a new release: https://github.com/apigee/apigee-edge-drupal/issues/302
